src/constbuffer.c

```diff
@@ -166,10 +213,15 @@ void CONSTBUFFER_Destroy(CONSTBUFFER_HANDLE constbufferHandle)
         /*Codes_SRS_CONSTBUFFER_02_016: [Otherwise, CONSTBUFFER_Destroy shall decrement the refcount on the constbufferHandle handle.]*/
         if (DEC_REF_VAR(constbufferHandle->count) == DEC_RETURN_ZERO)
         {
-            if (constbufferHandle->memory_moved)
+            if (constbufferHandle->buffer_type == CONSTBUFFER_TYPE_MEMORY_MOVED)
             {
                 free((void*)constbufferHandle->alias.buffer);
             }
+            else if (constbufferHandle->buffer_type == CONSTBUFFER_TYPE_WITH_CUSTOM_FREE)
+            {
+                /* Codes_SRS_CONSTBUFFER_01_012: [ If the buffer was created by calling CONSTBUFFER_CreateWithCustomFree, the customFreeFunc function shall be called to free the memory, while passed customFreeFuncContext as argument. ]*/
+                constbufferHandle->custom_free_func(constbufferHandle->custom_free_func_context);
```

Comment:
>custom_free_func [](start = 35, length = 16)

This could be null. Didn't see a check for it in the CONSTBUFFER_CreateWithCustomFree function. Did I miss something?